flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/sink/AsyncSinkBase.java

```diff
@@ -49,6 +50,28 @@
 public abstract class AsyncSinkBase<InputT, RequestEntryT extends Serializable>
         implements Sink<InputT, Void, Collection<RequestEntryT>, Void> {
 
+    protected final ElementConverter<InputT, RequestEntryT> elementConverter;
+    protected final int maxBatchSize;
+    protected final int maxInFlightRequests;
+    protected final int maxBufferedRequests;
+    protected final long flushOnBufferSizeInBytes;
+    protected final long maxTimeInBufferMS;
```

Comment:
Why are these protected rather than private?

Reply:
I would have thought the concrete implementation of the sink would need to access these values when creating the `SinkWriter`? e.g. here flink-connectors/flink-connector-aws/src/main/java/org/apache/flink/connector/kinesis/sink/KinesisDataStreamsSink.java on line 111

Reply:
Generally speaking I would be more inclined towards private variables with protected getters in this case. It is more maintainable for the future